include/stdexec/execution.hpp

```diff
@@ -5879,18 +5891,21 @@ namespace stdexec {
         return tag_invoke(sync_wait_t{}, (_Sender&&) __sndr);
       }
 
-      template <__single_value_variant_sender<__env> _Sender>
+      template <
+        __sync_wait::__driveable_context _Context,
+        __single_value_variant_sender<__env> _Sender>
         requires(!__tag_invocable_with_completion_scheduler< sync_wait_t, set_value_t, _Sender>)
              && (!tag_invocable<sync_wait_t, _Sender>) && sender_in<_Sender, __env>
-             && sender_to<_Sender, __receiver_t<_Sender>>
-      auto operator()(_Sender&& __sndr) const -> std::optional<__sync_wait_result_t<_Sender>> {
+             && sender_to<_Sender, __receiver_t<_Context, _Sender>>
+      auto operator()(_Context& __loop, _Sender&& __sndr) const
```

Comment:
I was imagining something more along the lines of:

```c++
template <sender _Sender, class _Env = empty_env>
   requires ...
auto operator()(_Sender&& __sndr, _Env&& __env = _Env()) 
```

... and then pull the driveable context out of `__env`, or use `run_loop` if it's not there. Eventually, we'll want to support other queries on the environment like a stop_token or allocator for the operation.

Also, I think `__single_value_variant_sender<__env>` is the wrong constraint. The `__env` type assumes a `run_loop` drivable context, no?

Reply:
I admit, today I've only changed the implementation of `io_context` to fix the hidden friend issue.

I imagine that the way you do. This was my previous attempt.

Regarding the `__single_value_variant_sender<__env>`, I believe it was wrong before my commit, too. Its a pending TODO. But I can fix that today or tomorrow.